core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -319,7 +313,7 @@ private void addInterceptor(AsyncInterceptorChain interceptorChain, AsyncInterce
    public void addPersistenceInterceptors(AsyncInterceptorChain interceptorChain, Configuration cacheConfiguration, List<StoreConfiguration> stores) {
       TransactionMode transactionMode = cacheConfiguration.transaction().transactionMode();
       CacheMode cacheMode = cacheConfiguration.clustering().cacheMode();
-      int position = persistenceSlot;
+      int position = calculatePersistenceSlot(interceptorChain);
```

Comment:
I was actually thinking of using `addInterceptorAfter` so that there is no window for other threads to change the EWI position between computing the index and actually adding the interceptors.

Reply:
I see, I will get rid of the index

Reply:
Changed to always use ```addInterceptorAfter```

Reply:
:+1:, sorry Tristan had to merge it Gustavo!